### PR TITLE
#1041 SMS Char Limit Changed to Warning

### PR DESCRIPTION
**Pull request by k-macmillan**
# Description

Update the SMS API validation exceeding character limit behavior so the HTTP response is 201 instead of 4xx.

The changes also include some clean-up recently made in other files:

    json.loads(response.get_text(...)) --> response.get_json()
    'sms' | 'email' | 'letter' --> SMS_TYPE | EMAIL_TYPE | LETTER_TYPE (which are constants defined in models.py)

#1041

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- [x] Unit tests pass
- [x] Manually sent a long SMS message, and verified a warning message appears in the logs.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by ianperera — DISMISSED** [review]

**Review by kalbfled — CHANGES_REQUESTED** [review]
These changes remove the tests that too-long requests return 4xx responses, but I think we need an affirmative test that too-long requests complete with a 201 response.

File: app/notifications/validators.py [hunk 1]
```diff
@@ -165,8 +159,11 @@ def validate_template(template_id, personalisation, service, notification_type):
     check_template_is_for_notification_type(notification_type, template.template_type)
     check_template_is_active(template)
     template_with_content = create_content_for_notification(template, personalisation)
-    if template.template_type == SMS_TYPE:
-        check_sms_content_char_count(template_with_content.content_count)
+    if template.template_type == SMS_TYPE and template_with_content.content_count > SMS_CHAR_COUNT_LIMIT:
+        current_app.logger.warning(
+            f"The personalized message length is {template_with_content.content_count}, which exceeds the "
+            "4 segments length of {SMS_CHAR_COUNT_LIMIT}."
```
Comment (kalbfled) [inline app/notifications/validators.py:165]: This needs "f" prepended.

File: app/notifications/validators.py [hunk 2]
```diff
@@ -160,10 +160,8 @@ def validate_template(template_id, personalisation, service, notification_type):
     check_template_is_active(template)
     template_with_content = create_content_for_notification(template, personalisation)
     if template.template_type == SMS_TYPE and template_with_content.content_count > SMS_CHAR_COUNT_LIMIT:
-        current_app.logger.warning(
-            f"The personalized message length is {template_with_content.content_count}, which exceeds the "
-            "4 segments length of {SMS_CHAR_COUNT_LIMIT}."
-        )
+        current_app.logger.warning("The personalized message length is %s, which exceeds the 4 segments length of %s.",
+                                   template_with_content.content_count, SMS_CHAR_COUNT_LIMIT)
```
Comment (kalbfled) [inline app/notifications/validators.py:164]: Good call.

**Review by cris-oddball — APPROVED** [review]

**Comment by k-macmillan** [comment]
This is tested in Dev and working as intended:
image

Messages under the limit do not leave that message (intended).

File: app/notifications/validators.py (hunk 1 above)
Comment (k-macmillan) [reply app/notifications/validators.py]: I switched it to printf style.

**Review by kalbfled — APPROVED** [review]

**Pull request merged**